### PR TITLE
Episode 16.2 - Weekend Memorial Dungeon

### PR DESCRIPTION
**Pull request by Everade**
* **Addressed Issue(s)**: 
https://github.com/rathena/rathena/issues/6451

* **Server Mode**: 
Renewal

* **Description of Pull Request**: 
Implementation of the Episode 16.2 - Weekend Memorial Dungeon.

### CODE REVIEW

**Comment by Everade** [comment]
Still ready for merge.

File: npc/re/instances/WeekendDungeon.txt [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+		mes "[Marry Jae]";
+		mes "Oh, a curious onlooker? It's Marry Jae, the person in charge of weekends among the days of the week.";
+		next;
+		mes "[Marry Jae]";
+		mes "What is the weekend dungeon? Actually, it's a specially designed space for new adventurers to train.";
+		next;
+		mes "^0000FFLooking around, it's obvious that Marry Jae is talking to you.^000000";
+		next;
+		mes "[Marry Jae]";
+		mes "For those who do not have a party on the weekends, this place are hunting and experience gaining grounds for lonely adventurers!";
```
Comment (aleos89) [inline npc/re/instances/WeekendDungeon.txt:29]:
```suggestion
		mes "For those who do not have a party on the weekends, this place is hunting and experience gaining grounds for lonely adventurers!";
```

File: npc/re/instances/WeekendDungeon.txt [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+						mes "Off you go!";
+						mapannounce "pay_arche", "" + strcharinfo(0) + " of the party, " + getpartyname( getcharid(1) ) + ", is entering the " + .@md_name$ + ".", bc_map, 0xFF99;
+						if (isbegin_quest(12377)<1) setquest 12377;
+						if (isbegin_quest(12378)<1) setquest 12378;
+				}
+				close;
+			case 3:
+				mes "[Marry Jae]";
+				mes "You have not prepared?";
+				mes "Can't really help it I guess";
```
Comment (aleos89) [inline npc/re/instances/WeekendDungeon.txt:95]:
```suggestion
				mes "Can't really help it I guess.";
```

File: npc/re/instances/WeekendDungeon.txt [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+		case 6:
+			.@min_level = 140;
+			setarray 'mob,3647,3641,3653; break;
+		case 7:
+			.@min_level = 160;
+			setarray 'mob,3648,3642,3654; break;
+	}
+	mes "[Marry Jae]";
+	if (BaseLevel < .@min_level) {
+		mes "Your base level is too low to select this, you need to be level "+.@min_level+"+";
```
Comment (aleos89) [inline npc/re/instances/WeekendDungeon.txt:185]:
```suggestion
		mes "Your base level is too low to select this, you need to be level "+.@min_level+"+.";
```

**Review by aleos89 — APPROVED** [review]
Some nit picks!

File: npc/re/instances/WeekendDungeon.txt [hunk 4]
```diff
@@ -0,0 +1,392 @@
...
+//= Weekend Memorial Dungeon
+//===== Description: =========================================
+//= [Walkthrough Conversion]
+//= Weekend Memorial Dungeon
+//===== Changelogs: ==========================================
+//= 1.0 Initial release [Everade]
+//============================================================
+
+pay_arche,44,124,5	script	Marry Jae	4_F_JOB_BLACKSMITH,{
+	mes "[Marry Jae]";
```
[the remark this answers is not in the document]
Comment (Everade) [reply npc/re/instances/WeekendDungeon.txt:11]: Possible, i made direct KRO translations, and that's really hard when it comes to names.
I think in iRO they called her "Mariage" for both Weekend and Friday Memorial Dungeon. Although the names were different in kRO.

**Pull request merged**